src/processor/block.processor.ts

```diff
@@ -0,0 +1,390 @@
+import { SyncDataHelpers } from 'src/helpers/sync-data.helpers';
+import { TransactionHelper } from 'src/helpers/transaction.helper';
+import { BlockSyncErrorRepository } from 'src/repositories/block-sync-error.repository';
+import { SmartContractRepository } from 'src/repositories/smart-contract.repository';
+import { SyncStatusRepository } from 'src/repositories/sync-status.repository';
+import { ENV_CONFIG } from 'src/shared/services/config.service';
+import { CommonUtil } from 'src/utils/common.util';
+import { InfluxDBClient } from 'src/utils/influxdb-client';
+import { In } from 'typeorm';
+import { ProposalVoteRepository } from 'src/repositories/proposal-vote.repository';
+import { Logger } from '@nestjs/common';
+
+@Processor(QUEUES.SYNC_BLOCK.QUEUE_NAME)
+export class BlockProcessor {
+  private readonly logger = new Logger(BlockProcessor.name);
+  private threads = 0;
+  private influxDbClient: InfluxDBClient;
+  private api = ENV_CONFIG.NODE.API;
+
+  constructor(
+    private commonUtil: CommonUtil,
+    private blockSyncErrorRepository: BlockSyncErrorRepository,
+    private statusRepository: SyncStatusRepository,
+    private smartContractRepository: SmartContractRepository,
+    private proposalVoteRepository: ProposalVoteRepository,
+
+    @InjectQueue(QUEUES.SYNC_BLOCK.QUEUE_NAME)
+    private readonly blockQueue: Queue,
+    @InjectQueue(QUEUES.SYNC_CONTRACT.QUEUE_NAME)
+    private readonly contractQueue: Queue,
+  ) {
+    this.logger.log(
+      '============== Constructor Block Processor ==============',
+    );
+
+    this.threads = 100;
+
+    this.blockQueue.add(
+      QUEUES.SYNC_BLOCK.JOBS.SYNC_BLOCK_HEIGHT,
+      {},
+      { removeOnFail: false, repeat: { every: Number(ENV_CONFIG.TIMES_SYNC) } },
+    );
+
+    this.blockQueue.add(
+      QUEUES.SYNC_BLOCK.JOBS.PROCESS_BLOCK,
+      {},
+      { removeOnFail: false, repeat: { every: 3000 } },
+    );
+
+    this.connectInfluxDB();
+  }
+
+  @Process(QUEUES.SYNC_BLOCK.JOBS.SYNC_BLOCK_HEIGHT)
+  async syncBlockHeight() {
+    // Get the highest block and insert into SyncBlockError
+    const blockErrors = [];
+    try {
+      let currentHeight = 0;
+      this.logger.log('start cron generate block sync error');
+      const [blockLatest, currentBlock, blockStatus] = await Promise.all([
+        this.getBlockLatest(),
+        this.blockSyncErrorRepository.max('height'),
+        this.statusRepository.findOne(),
+      ]);
+
+      const height = Number(currentBlock?.height);
+      const currentStatusBlock = Number(blockStatus?.current_block);
+
+      this.logger.log(`Current block height: ${height}`);
+      this.logger.log(`Current block status: ${currentStatusBlock}`);
+
+      if (height > currentStatusBlock) {
+        currentHeight = Number(currentBlock.height);
+      } else {
+        currentHeight = Number(blockStatus.current_block) || 0;
+      }
+
+      let latestBlk = Number(blockLatest?.block?.header?.height || 0);
```

Comment:
Change name có nghĩa hơn: 
```suggestion
      let latestBlockNumber = Number(blockLatest?.block?.header?.height || 0);
```